term-utils/agetty.c

```diff
@@ -1507,23 +1508,6 @@ static void auto_baud(struct termios *tp)
 	tcsetattr(STDIN_FILENO, TCSANOW, tp);
 }
 
-static char *xgethostname(void)
```

Comment:
Please, don't add agetty changes to this PR :-) BTW, agetty uses local xgethostname() due to log_err(); this function writes to syslog.

Reply:
The changes were made because I included "xmalloc" in "all-io.h", which caused a duplicate declaration error during compilation. However, I have now reverted those changes as they were not appropriate.

Thank you for providing the background information.